File: 1 lab/main.py
list = []
print("Enter a number, if you want to stop, press enter:")
while True:
    try:
      a = input('-->> ')
      if a == '':
          break
      a = float(a)
      print("This is the correct input! Your number: ", a)
      list.append(a)
    except ValueError:
        print("This is an incorrect input. It's not a number, try again.")
print("Your collection: ", list)

sum = 0
for number in list:
    if number > 0:
        sum += number
print("The sum of the positive elements of your collection: ", sum)

File: 1 lab/main.cpp
#include <iostream>
using namespace std;

int InpInt(){
    int n;
    while (true) {
    		cin >> n;
    		if (cin.get() == '\n') return n;
    		else {
    		    cin.clear();
    		    cin.ignore(256, '\n');
    			cout << "Error! Try again" << endl;
    		}
    	}
}

int main()
{
	int n; 
	cout << "Enter the size of the array: ";
	n = InpInt();
	double sum = 0;
	double arr[n];
	for(int i=0; i<n; i++)
	{
	    cout << "Enter values: ";
		try
        {
            double a;
            do
            {
                if(!cin)
                {
                    cin.clear();
                    while (cin.get() != '\n');
                    cout<<"Error! try again\n";
                }
                cin>>a;
            }while(!cin);
            arr[i] = a;
        }
        catch(...)
        {
            cout << "Unknown error! \n";
        }
		if (arr[i] > 0){
		  sum += arr[i];
		}
	}
	cout << "  The sum of positive numbers: " << sum;
	return 0;
}
